src/ngraph/runtime/gpu/gpu_kernel_args.cpp

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+* Copyright 2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <string>
+
+#include "ngraph/runtime/gpu/gpu_kernel_args.hpp"
+
+#define TI(x) std::type_index(typeid(x))
+
+using namespace ngraph;
+
+const std::unordered_map<std::type_index, std::string> runtime::gpu::GPUKernelArgs::type_names = {
+    {TI(size_t), "size_t"},
+    {TI(char), "char"},
+    {TI(float), "float"},
+    {TI(double), "double"},
+    {TI(int8_t), "int8_t"},
+    {TI(int16_t), "int16_t"},
+    {TI(int32_t), "int32_t"},
+    {TI(int64_t), "int64_t"},
+    {TI(uint8_t), "uint8_t"},
+    {TI(uint16_t), "uint16_t"},
+    {TI(uint32_t), "uint32_t"},
+    {TI(uint64_t), "uint64_t"},
+    {TI(int), "int"},
```

Comment:
Aren't these old style types covered in int32_t, etc.? If not do we need short and unsigned short, char and unsigned char?

Reply:
I checked, and they are covered.